dspace-server-webapp/src/main/java/org/dspace/app/rest/ProcessFilesRestController.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;
+
+import java.sql.SQLException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.app.rest.link.HalLinkService;
+import org.dspace.app.rest.link.process.ProcessResourceHalLinkFactory;
+import org.dspace.app.rest.model.BitstreamRest;
+import org.dspace.app.rest.model.ProcessRest;
+import org.dspace.app.rest.model.hateoas.BitstreamResource;
+import org.dspace.app.rest.repository.ProcessRestRepository;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.authorize.AuthorizeException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PagedResourcesAssembler;
+import org.springframework.hateoas.Link;
+import org.springframework.hateoas.PagedModel;
+import org.springframework.hateoas.server.mvc.WebMvcLinkBuilder;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/" + ProcessRest.CATEGORY + "/" + ProcessRest.PLURAL_NAME + "/{processId}/files")
+public class ProcessFilesRestController {
```

Comment:
A small update was needed on the RestResourceController in order to support one of the 2 methods for which this controller was needed. The remaining one is too specific & cannot be moved to the repository (unless we change the contract & this is still an open discussion).

Reply:
@KevinVdV : I don't understand your note here, as the path this Controller responds to is not in the Contract: https://github.com/DSpace/Rest7Contract/pull/99   I'd need to understand the purpose of this endpoint as it's not clear to me.

Also, it's worth noting, per the [REST Contract README](https://github.com/DSpace/Rest7Contract/blob/master/README.md), we don't support subpaths of the type you've added here.  You seem to be implying by the path `/process/[id]/files/name/[name]` that there are TWO unique identifiers for a File within a Process, either the `name` or the `type` (which is already documented as the unique identifier in the contract).  This is something we've not allowed on other endpoints, so we'd need a different path here that doesn't require a custom Controller (Again, as I've noted, we highly recommend against custom Controllers. If you find yourself creating one, please ask for advice on how to avoid it.)  One manner of implementing this in another way would be to reconsider this as a `/api/core/bitstreams/search/byProcessAndName?process=[]&name=[]`.  It could still end up with a unique result, but it'd clarify that it's not the main unique identifier (and that Type is considered the main unique identifier).

Reply:
This controller was removed as the name endpoint isn't used. A small leftover from an initial implementation.